### PR TITLE
rollback incompatible cross imports

### PR DESCRIPTION
**Pull request by dmarticus**
I accidentally included an import in https://github.com/higherkindness/skeuomorph/pull/321 that doesn't compile correctly in scala 2.12.  My bad.  This change should fix that.

### CODE REVIEW

**Comment by dmarticus** [comment]
Does anyone know how to instrument cross-compiled tests for repos like this one?  I was able to accidentally break `sbt-mu-srcgen` @ Scala 2.12 by accidentally upgrading to some scala 2.13 features that should've been caught by compiling the codebase with 2.12.  Could a github action be modified, or something?

**Comment by fedefernandez** [comment]
@dmarticus that's weird. The project is cross-compiled against Scala 2.12

https://github.com/higherkindness/skeuomorph/runs/996164886#step:10:246

Maybe we're excluding the problematic files from that test execution?

File: src/test/scala/higherkindness/skeuomorph/protobuf/ProtobufProtocolSpec.scala [hunk 1]
```diff
@@ -56,7 +58,7 @@ class ProtobufProtocolSpec extends Specification with ScalaCheck {
   The generated Scala code should escape 'type' keyword in package (directory) names. $codegenGoogleApi
   """
 
-  def codegenProtobufProtocol =
+  def codegenProtobufProtocol: ScalaCheckFunction2[CompressionType, Boolean, Prop] =
```
Comment (pepegar) [inline src/test/scala/higherkindness/skeuomorph/protobuf/ProtobufProtocolSpec.scala:61]: Isn´t the compiler able to infer this?

**Review by pepegar — APPROVED** [review]
LGTM, just one minor comment.

**Comment by cb372** [comment]
I'm a bit confused by this PR.

The project has a dependency on `scala-collection-compat`, so `import scala.jdk.CollectionConverters._` will compile just fine?

File: src/test/scala/higherkindness/skeuomorph/protobuf/ProtobufProtocolSpec.scala (hunk 1 above)
Comment (dmarticus) [reply src/test/scala/higherkindness/skeuomorph/protobuf/ProtobufProtocolSpec.scala:61]: yeah it was; this was a leftover change to me monkeying around with a test; I'll close it.

**Comment by dmarticus** [comment]
@cb372 sorry I submitted when my brain was a bit fried and I was worried I'd broken something.  Here's the train of thought that I went through that made me think it was necessary.  

1. I made a local change to skeuomorph to see if I was correctly handling optional packages in the import files (I'm working on this issue: https://github.com/higherkindness/skeuomorph/issues/318) and I published it locally to see if my changes would plumb all the way through the `sbt-mu-srcgen`.  Since `sbt-mu-srcgen` uses scala 2.12, I published locally using `sbt ++2.12.11 publishLocal`.

2. when I used the locally published version of skeuomorph (`v0.0.24+19-88f86878+20200819-1630-SNAPSHOT`), `sbt-mu-srcgen` couldn't compile, and it had a bunch of errors related to the collector lib:
```
[info] higherkindness.mu.rpc.srcgen.ProtoSrcGenTests *** ABORTED ***
[info]   java.lang.NoSuchMethodError: 'scala.collection.convert.Decorators$AsScala scala.jdk.CollectionConverters$.asScalaBufferConverter(java.util.List)'
[info]   at higherkindness.skeuomorph.protobuf.ParseProto$.$anonfun$getTFiles$1(ParseProto.scala:120)
[info]   at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:87)
[info]   at cats.effect.internals.IORunLoop$RestartCallback.signal(IORunLoop.scala:366)
[info]   at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:387)
[info]   at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:330)
[info]   at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:141)
[info]   at cats.effect.internals.IORunLoop$.startCancelable(IORunLoop.scala:41)
[info]   at cats.effect.internals.IOBracket$BracketStart.run(IOBracket.scala:90)
[info]   at cats.effect.internals.Trampoline.cats$effect$internals$Trampoline$$immediateLoop(Trampoline.scala:67)
[info]   at cats.effect.internals.Trampoline.startLoop(Trampoline.scala:35)
[info]   ...
```

Based on that error, I assumed that it had something to do with the cross-compiling.  If that's not the case, I'd be happy to close this PR, but I would be curious on how I should publish any `skeuomorph` changes locally to work with `sbt-mu-srcgen`, since the version published by my `sbt ++2.12.11 publishLocal` command doesn't compile.

cc @pepegar you might be curious about this, too.

**Comment by cb372** [comment]
I should have time to investigate this today

**Comment by cb372** [comment]
OK I worked out what is going on in sbt-mu-srcgen.

The problem is avrohugger. It defines its own `scala.jdk.CollectionConverters` object [here](https://github.com/julianpeeters/avrohugger/blob/844109d6933f035714e440841bed6091db931bbc/avrohugger-core/src/main/scala-2.12/scala/jdk/package.scala), which means the object with the same name provided by scala-collection-compat is invisible.

I tried adding scala-collection-compat to the `libraryDependencies` of sbt-mu-srcgen, in order to force the classpath ordering. That fixes (in a very fragile way) the error you saw, but it causes a similar runtime error in the Avro tests because avrohugger's `scala.jdk.CollectionConverters` object gets shadowed:

```
[info]   NoSuchMethodError was thrown during property evaluation.
[info]     Message: scala.jdk.CollectionConverters$.RichConcurrentHashMap(Ljava/util/concurrent/ConcurrentHashMap;)Lscala/jdk/CollectionConverters$RichConcurrentHashMap;
```

In conclusion:
1. For now, let's avoid use of `scala.jdk.CollectionConverters` in skeuomorph, i.e. I agree with what you're doing in this PR. We'll just have to live with the deprecation warnings on 2.13.
2. Longer term, let's try to move away from avrohugger, or perhaps fork it and get rid of the home-grown `scala.jdk.CollectionConverters` object

**Comment by dmarticus** [comment]
@cb372 thanks for your investigation!  I can add a comment to this PR explaining why we're getting the warnings, or I can just merge it as is.  This change is also handled in https://github.com/higherkindness/skeuomorph/pull/324, so I can just close this change in favor of 324 instead, too.

**Comment by dmarticus** [comment]
I'm gonna close this in favor of #324 because that change is basically ready for merge and it satisfies the codecov bot ;)
